### ISSUE TITLE
formal BNF description of demes yaml format

### ISSUE THREAD

**Issue by grahamgower** [grahamgower/demes#56]
We should write an BNF/[EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form) description of the declarative file format. If should be fairly short, because we don't have so much syntax or many keywords. At the very least, it will be nice to have a complete description to include in the documentation. BNF is machine readable though, and there are many tools that can do useful things with this (and the fuzzer suggested in #24 could consume this format).

**Comment by jeromekelleher** [comment]
That would be awesome, if it's possible. I've never been any good with BNF though, so won't be of any help.

**Comment by grahamgower** [comment]
BNF always used to confuse me, but I can sorta wrap my head around it nowadays. Writing something up should clear out any remaining cobwebs!

**Comment by grahamgower** [comment]
I'm definitely not going to do this. YAML is crazy complicated, and doing this is far beyond useful. I'm just going to show the `strictyaml` schema in the docs.

**Comment by jeromekelleher** [comment]
> I'm definitely not going to do this. YAML is crazy complicated, and doing this is far beyond useful. I'm just going to show the strictyaml schema in the docs.

Definitely! No one's going to use a BNF description anyway, we'd be better off providing actual parsers.
